src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/CleanUpRejectedFilesTask.java

```diff
@@ -53,58 +56,67 @@ public void run() {
+            .map(blobItem -> containerClient.getBlobClient(blobItem.getName()))
+            .forEach(blobClient -> leaseAcquirer.ifAcquiredOrElse(
+                blobClient,
+                leaseId -> delete(blobClient, leaseId),
+                errorCode -> {}, // nothing to do if blob not found in rejected container
+                false
+            ));
+
+        log.info("Finished removing rejected files. Container: {}", containerName);
+    }
+
+    private void delete(BlobClient blobClient, String leaseId) {
         try {
-            CloudBlockBlob blob = container.getBlockBlobReference(
-                zipFileName
+            blobClient.deleteWithResponse(
+                DeleteSnapshotsOptionType.INCLUDE,
+                new BlobRequestConditions().setLeaseId(leaseId),
+                null,
+                Context.NONE
             );
-
-            if (canBeDeleted(blob)) {
-                Optional<String> leaseId = blobManager.acquireLease(
-                    blob,
-                    container.getName(),
-                    zipFileName
-                );
-                if (leaseId.isPresent()) {
-                    AccessCondition accessCondition = generateLeaseCondition(leaseId.get());
-                    blob.delete(INCLUDE_SNAPSHOTS, accessCondition, null, null);
-                    log.info(
-                        "Deleted rejected file {} from container {}",
-                        blob.getName(),
-                        container.getName()
-                    );
-                }
-            }
-        } catch (URISyntaxException | StorageException exc) {
+            log.info(
+                "Deleted rejected file {} from container {}",
+                blobClient.getBlobName(),
+                blobClient.getContainerName()
+            );
+        } catch (Exception ex) {
             log.error(
                 "Unable to delete rejected file {} from container {}",
-                zipFileName,
-                container.getName(),
-                exc
+                blobClient.getBlobName(),
+                blobClient.getContainerName(),
+                ex
             );
         }
     }
 
-    private boolean canBeDeleted(CloudBlockBlob blob) throws StorageException {
-        blob.downloadAttributes();
-
-        Date createdTime = blob.getProperties().getLastModified();
-        Date cutoff = Date.from(Instant.now().minus(this.ttl));
-
-        return createdTime.before(cutoff);
+    //ToDo remove log
+    private boolean canBeDeleted(BlobItem blobItem) {
+        OffsetDateTime lastModified = blobItem.getProperties().getLastModified();
+        OffsetDateTime now = now(UTC);
+        OffsetDateTime lastTimeToStay = lastModified.plus(ttl);
+        log.info(
+            "Blob's last modified: {}, now time: {}, lastTimeToStay: {}",
+            lastModified,
+            now,
+            lastTimeToStay
+        );
+        return lastTimeToStay
+            .isBefore(now(UTC));
```

Comment:
No reason to break this line

Reply:
Any reason for using lastTimeToStay instead of cutoff?

Reply:
types are different that is why this part changed. It is just for logging after I check the logs I will remove the `vars`.